docs: add internship call blog post (#141)
OpenScienceLabs/opensciencelabs.github.io: pages/blog/internship-call-2-2024/index.md

  }

  td {
    background-color: #d8f3dc;
  }

  tr:hover td {
    background-color: #f2f2f2;
  }

  pre {
    background-color: #f8f8f8;
    padding: 10px;
    border: 1px solid #ddd;
    overflow-x: auto;
  }

  a {
    color: #3498db;
    text-decoration: none;
  }

  a:hover {
    text-decoration: underline;
  }
</style>

<!-- # Intro -->

This program presents valuable opportunities for both mentors and student/collaborators to engage and grow, despite being an unpaid initiative.

## Summary of steps to got into internship program

- Choose a project of your interest. Please read the project ideas document and see if yout meet the prerequisites or the stack required (normally the mentor rejects applicants if they do not meet the stack required)
- Contact to the project mentor through email with the following info.
    - Curriculum Vitae
    - The project name
    - Project idea name
    - Motivation for wanting to collaborate on that specific project.
- The mentor will assign you a minor issue to solve. (This step is required to verify if the applicant is able to solve the project tasks.)
- After solving the issue The mentor will ask you to fill a form via google forms.

> - The selected applicant will be contacted via email on May 20 week.

Reviewer: ```suggestion
- The selected applicant will be contacted via email on May 20.
```

> Candidates are reminded of the importance of having prior knowledge of Git and how to make contributions we recommend this read this [blog post](https://opensciencelabs.org/blog/first-time-contributors/). Additionally, candidates should join the [OSL Discord](https://opensciencelabs.org/discord) to stay updated on announcements related to the Internship Program. We recommend to read this blog post for first time contributions

Reviewer: ```suggestion
Candidates should have a basic understanding of Git and how to contribute. For guidance, we recommend reading this [blog post](https://opensciencelabs.org/blog/first-time-contributors/) designed for first-time contributors. Additionally, to stay informed about announcements related to the Internship Program, candidates are encouraged to join the [OSL Discord](https://opensciencelabs.org/discord).
```

    padding: 12px;
    text-align: left;
    transition: background-color 0.3s;
    color: #000; /* Cambio del color del texto a negro */
  }

  th {
    background-color: #74c69d; /* Color del título */
    color: #000;
  }

  td {
    background-color: #d8f3dc;
  }

  tr:hover td {
    background-color: #f2f2f2;
  }

  pre {
    background-color: #f8f8f8;
    padding: 10px;
    border: 1px solid #ddd;
    overflow-x: auto;
  }

  a {
    color: #3498db;
    text-decoration: none;
  }

  a:hover {
    text-decoration: underline;
  }
</style>

<!-- # Intro -->

This program presents valuable opportunities for both mentors and student/collaborators to engage and grow, despite being an unpaid initiative.

## Summary of steps to got into internship program

- Choose a project of your interest. Please read the project ideas document and see if yout meet the prerequisites or the stack required (normally the mentor rejects applicants if they do not meet the stack required)
- Contact to the project mentor through email with the following info.
    - Curriculum Vitae
    - The project name
    - Project idea name
    - Motivation for wanting to collaborate on that specific project.

> - The mentor will assign you a minor issue to solve. (This step is required to check if the applicant knows git)

Reviewer: actually, it is not just about git, this step is important to check if the candidate would be able to accomplish with the tasks in the proposal.

it could mean knowledge about git/github, ability to set up and run the project locally, ability to navigate the project sources, execute pre-commit, etc and ability to identify issues pointed locally or CI (although sometime with help from the mentors), etc

At the end, we can also point to this first time contributor guide: https://opensciencelabs.org/blog/first-time-contributors/

> - After solving the issue The mentor will ask you to fill a form via LinkedIn

Reviewer: Although the linkedin idea would be great, but as we have the limit of 25 applicants, I think that a lot of candidates will try to fill that without being approved by the mentors, and we will run out of spots there in the earlier stage of the application period.

for now, maybe we could have that same form we have for linkedin but in google forms structure

- The seleted applicant will be contated via email on May 20 week

Candidates are reminded of the importance of having prior knowledge of Git. Additionally, candidates should join the [OSL Discord](https://opensciencelabs.org/discord) to stay updated on announcements related to the Internship Program.

> Do not forget about the timeline and if you have any more questions do not hesitate to contact us through opensciencelabs@gmail.com

Reviewer: ```suggestion
Please keep in mind the timeline, and if you have any more questions, do not hesitate to contact us at team@opensciencelabs.org.
```

## Timeline

> The following is the timeline for the OSL Internship Program Cycle 2024-01:

Reviewer: ```suggestion
Here is the timeline for the OSL Internship Program Cycle 2024-02:
```

> - **April, 22**: Call for Interns/Apprentices opens.

Reviewer: ```suggestion
- **April 22**: Call for Interns/Apprentices opens.
```

> - **May, 13**: Deadline for Interns/Apprentices applications.

Reviewer: ```suggestion
- **May 13**: Deadline for Interns/Apprentices applications.
```

> - **May, 20**: Announcement of approved Interns/Apprentices and start of bounding period

Reviewer: ```suggestion
- **May 20**: Announcement of approved Interns/Apprentices and start of the bonding period.
```

> - **May, 27**: Official Start Date of Internship Period; an alternative for projects not 

Reviewer: ```suggestion
- **May 27**: Official Start Date of the Internship Period; an alternative for projects not 
```

selected by GSoC to run under the OSL Internship Program with The Graph Network support.

> - **July, 8**: Mid-term Evaluation.

Reviewer: ```suggestion
- **July 8**: Mid-term Evaluation.
```

> - **August, 26**: Final Evaluation.

Reviewer: ```suggestion
- **August 26**: Final Evaluation.
```

> - **September, 3**: Official End Date; Certification process begins.

Reviewer: ```suggestion
- **September 3**: Official End Date; Certification process begins.
```

## Info about participating Projects
---

> Below is the list of projects participating in the current internship cycle. Each project includes key details to help candidates understand the scope and requirements.

Reviewer: ```suggestion
Below is the list of projects participating in the current internship cycle. Each entry provides key details to assist candidates in understanding the scope and requirements of the projects.
```

- **Categories:** web development, backend, frontend, python
- **Organization/Project Webpage URL**: <https://github.com/osl-incubator/growth-forge/>
- **Contact**: Ivan Ogasawara (ivan.ogasawara@gmail.com)
- **Project Ideas URL**: <https://github.com/osl-incubator/growth-forge/wiki/OSL-Internship-%E2%80%90-2024-%E2%80%90-2nd-Cycle>

### Useful Materials and Courses
---

- **Software Carpentry Lessons:** Offering tutorials on Git, Bash, Python, R, and more, these lessons are invaluable for building a strong foundation in software development. Access the lessons at Software Carpentry.
- **Udacity CS212 - Design of Computer Programs:** This course, taught by Peter Norvig, delves into advanced programming topics and is an excellent way to deepen your understanding of computer programs. Enroll in the course at Udacity CS212.
- **The GRAPH Network Courses:** Explore a range of courses offered by The GRAPH Network, tailored to various aspects of data analysis. Find the courses at The GRAPH Network Courses.

> These resources provide a great opportunity to prepare effectively for the Internship Program and to develop a broad skill set in software development and data analysis.

Reviewer: it would be nice to add to this list a link to: https://opensciencelabs.org/blog/first-time-contributors/
